grblas/tests/test_matrix.py

```diff
@@ -2103,3 +2105,86 @@ def test_concat(A):
 
 def test_nbytes(A):
     assert A.ss.nbytes > 0
+
+
+def test_auto(A, v):
+    expected = binary.times(A & A).new()
+    for expr in [(A & A), binary.times(A & A)]:
+        assert expr.dtype == expected.dtype
+        assert expr.nrows == expected.nrows
+        assert expr.ncols == expected.ncols
+        assert expr.shape == expected.shape
+        assert expr.nvals == expected.nvals
+        assert expr.isclose(expected)
+        assert expected.isclose(expr)
+        assert expr.isequal(expected)
+        assert expected.isequal(expr)
+        assert expr.mxv(v).isequal(expected.mxv(v))
+        assert expected.T.mxv(v).isequal(expr.T.mxv(v))
+        for method in [
+            "ewise_add",
+            "ewise_mult",
+            "mxm",
+            "__matmul__",
+            "__and__",
+            "__or__",
+            "kronecker",
+        ]:
+            val1 = getattr(expected, method)(expected).new()
+            val2 = getattr(expected, method)(expr)
+            val3 = getattr(expr, method)(expected)
+            val4 = getattr(expr, method)(expr)
+            assert val1.isequal(val2)
+            assert val1.isequal(val3)
+            assert val1.isequal(val4)
+        for method in ["reduce_rows", "reduce_columns", "reduce_scalar"]:
+            s1 = getattr(expected, method)().new()
+            s2 = getattr(expr, method)()
+            assert s1.isequal(s2.new())
+            assert s1.isequal(s2)
+
+    expected = semiring.plus_times(A @ v).new()
+    for expr in [(A @ v), (v @ A.T), semiring.plus_times(A @ v)]:
+        assert expr.vxm(A).isequal(expected.vxm(A))
+        assert expr.vxm(A).new(mask=expr.S).isequal(expected.vxm(A).new(mask=expected.S))
+        assert expr.vxm(A).new(mask=expr.V).isequal(expected.vxm(A).new(mask=expected.V))
+
+
+def test_expr_is_like_matrix(A):
+    attrs = {attr for attr, val in inspect.getmembers(A)}
+    expr_attrs = {attr for attr, val in inspect.getmembers(binary.times(A & A))}
+    infix_attrs = {attr for attr, val in inspect.getmembers(A & A)}
+    transposed_attrs = {attr for attr, val in inspect.getmembers(A.T)}
+    # Should we make any of these raise informative errors?
+    expected = {
```

Comment:
This is a good place to look for whether expressions "look like" the original object.